core/src/main/scala/cats/data/NonEmptyList.scala

```diff
@@ -106,6 +107,20 @@ final case class NonEmptyList[A](head: A, tail: List[A]) {
     toList.iterator.map(A.show).mkString("NonEmptyList(", ", ", ")")
 
   override def toString: String = s"NonEmpty$toList"
+
+  /**
+   * Remove duplicates. Duplicates are checked using `Order[_]` instance.
+   */
+  def distinct(implicit O: Order[A]): NonEmptyList[A] = {
```

Comment:
only question: there is an `O(N^2)` algorithm that uses only `Eq[A]` I wonder is it makes sense to also implement that?

we could use something like: https://github.com/non/algebra/blob/master/core/src/main/scala/algebra/Priority.scala

(or we could use `Xor` for that) and have something like:
```scala
def distinct(implicit oe: Xor[Eq[A], Order[A]]): NonEmptyList[A] = oe match {
  case Xor.Right(ord) => // do the tree set which is `O(log N)` per check
  case Xor.Left(eq) => // do a "listset" approach of checking each item, this incurs `O(N)` per check
}
```
If we had a `Hash[A]` type that potentially extended `Eq[A]` we could even have something like:
```scala
def distinct(implicit oe: Xor[Eq[A], Xor[Order[A], Hash[A]]]): NonEmptyList[A]
```
which would prefer to use hash sets, then tree sets, then list sets.

This is perhaps best served with different method names so readers can be more clear which complexity they get, but it is interesting that the semantics of the method don't care.